src/main/mondrian/olap/fun/FilterFunDef.java

```diff
@@ -114,12 +115,20 @@ public TupleIterable evaluateIterable(Evaluator evaluator) {
                 NativeEvaluator nativeEvaluator =
                     schemaReader.getNativeSetEvaluator(
                         call.getFunDef(), call.getArgs(), evaluator, this);
+                TupleIterable result;
                 if (nativeEvaluator != null) {
-                    return (TupleIterable)
-                        nativeEvaluator.execute(ResultStyle.ITERABLE);
+                    try {
+                        result = (TupleIterable) nativeEvaluator
+                            .execute(ResultStyle.ITERABLE);
+                    } catch (NativeEvaluationInClauseLimitException e) {
+                        // "fallback" strategy supposes
+                        // mondrian.native.unsupported.alert equals "ERROR"
+                        result = makeIterable(evaluator);
+                    }
                 } else {
```

Comment:
@YuryBY I was imagining that we would determine up-front that the constraint cannot be handled natively, probably by adding a check to isValidContext.  I believe we could exceed maxConstraints for the context constraint with other native evaluators, as well, so it would be preferable to check for validity and not construct a nativeEvaluator, rather than attempting and throwing an exception.

I had left a comment in the case about how we might be able to do that check.